pulp_python/app/viewsets.py

```diff
@@ -17,6 +41,42 @@ class PythonPackageContentViewSet(platform.ContentViewSet):
     queryset = python_models.PythonPackageContent.objects.all()
     serializer_class = python_serializers.PythonPackageContentSerializer
 
+    @transaction.atomic
+    def create(self, request):
+        try:
+            artifact = self.get_resource(request.data['artifact'], Artifact)
+        except KeyError:
+            raise serializers.ValidationError(detail={'artifact': _('This field is required')})
+
+        filename = request.data['filename']
+
+        # iterate through extensions since splitext does not support things like .tar.gz
+        for ext, packagetype in DIST_EXTENSIONS.items():
+            if filename.endswith(ext):
+                with tempfile.TemporaryDirectory() as td:
+                    temp_path = os.path.join(td, filename)
+                    shutil.copy2(artifact.file.path, temp_path)
+                    metadata = DIST_TYPES[packagetype](temp_path)
+                    metadata.packagetype = packagetype
+                    break
+        else:
+            raise ValueError(_("Filename extension is not a known python extension."))
+
+        data = parse_project_metadata(metadata.__dict__)
```

Comment:
would `parse_project_metadata(dict(metadata))` work? I think that would be more pythonic.

Reply:
I don't think you can call `dict()` on a object to get its attributes. But I will use `vars()` instead which looks a lot nicer than using `object.__dict__`